test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -6287,6 +6287,26 @@ BOOST_AUTO_TEST_CASE(divisiod_by_zero)
 	BOOST_CHECK(callContractFunction("mod(uint256,uint256)", 7, 0) == encodeArgs());
 }
 
+BOOST_AUTO_TEST_CASE(sdiv_minint_by_negone)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function minInt() returns (int) {
+				return -2**255;
+			}
+			function div(int a, int b) returns (int) {
+				return a / b;
+			}
+			function test() returns (int) {
+				return div(minInt(), -1);
```

Comment:
I understand the need of `minInt()`, otherwise it would be a literal number and hence won't become assembly.

I think `return minInt() / -1;` shouldn't be optimised out, but changing `mintInt` to `external` will definitely not let the optimiser take over. Maybe this is something we need to consider for tests.